Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.DataGridViewRow.cs

```diff
@@ -0,0 +1,710 @@
+        /// </summary>
+        /// 
+        /// <returns>
+        /// A <see cref="T:System.Windows.Forms.DataGridViewCellCollection"/> that contains all of the cells in the row.
+        /// </returns>
+        public DataGridViewCellCollection Cells
+        {
+            get
+            {
+                Contract.Ensures(Contract.Result<DataGridViewCellCollection>() != null);
+                return default(DataGridViewCellCollection);
+            }
+        }
+
+        // <summary>
+        // Gets or sets the shortcut menu for the row.
+        // </summary>
+        // 
+        // <returns>
+        // The <see cref="T:System.Windows.Forms.ContextMenuStrip"/> associated with the current <see cref="T:System.Windows.Forms.DataGridViewRow"/>. The default is null.
+        // </returns>
+        // <exception cref="T:System.InvalidOperationException">When getting the value of this property, the row is in a <see cref="T:System.Windows.Forms.DataGridView"/> control and is a shared row.</exception>
+        // public override ContextMenuStrip ContextMenuStrip {get; set;}
+        
+        // <summary>
+        // Gets the data-bound object that populated the row.
+        // </summary>
+        // 
+        // <returns>
+        // The data-bound <see cref="T:System.Object"/>.
+        // </returns>
+        // public object DataBoundItem { get;  }
+        
+        /// <summary>
+        /// Gets or sets the default styles for the row, which are used to render cells in the row unless the styles are overridden.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// The <see cref="T:System.Windows.Forms.DataGridViewCellStyle"/> to be applied as the default style.
+        /// </returns>
+        /// <exception cref="T:System.InvalidOperationException">When setting this property, the row is in a <see cref="T:System.Windows.Forms.DataGridView"/> control and is a shared row.</exception><PermissionSet><IPermission class="System.Security.Permissions.EnvironmentPermission, mscorlib, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Unrestricted="true"/><IPermission class="System.Security.Permissions.FileIOPermission, mscorlib, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Unrestricted="true"/><IPermission class="System.Security.Permissions.SecurityPermission, mscorlib, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Flags="UnmanagedCode, ControlEvidence"/><IPermission class="System.Diagnostics.PerformanceCounterPermission, System, Version=2.0.3600.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" version="1" Unrestricted="true"/></PermissionSet>
+        public DataGridViewCellStyle DefaultCellStyle
+        {
+            get { return default(DataGridViewCellStyle); }
+            set
+            {
+                Contract.Requires(!(this.DataGridView != null && this.Index == -1));
```

Comment:
Please use regular pattern for this: `Requires(DataGridView == null || Index != -1);`.